schemas/JSON/manifests/v1.2.0/manifest.installer.1.2.0.json

```
@@ -466,9 +497,7 @@
           "$ref": "#/definitions/Scope"
         },
         "InstallerUrl": {
-          "type": "string",
-          "pattern": "^([Hh][Tt][Tt][Pp][Ss]?)://.+$",
-          "maxLength": 2048,
+          "$ref": "#/definitions/Url",

```

>"$ref": "#/definitions/Url", [](http://example.com/codeflow?start=10&length=28)

This url definition allows null, for InstallerUrl, we should keep the original definition which does not allow null

---

```suggestion
          "allOf": [
            {"$ref":"#/definitions/Url"},
            {"type":"string"}
          ],
```
This keeps the validation that it matches the URL parameter, but adds the restriction that it must be a string, without duplicating URL logic

---

I decided to go with Trenly's latter suggestion in order to ensure that installerUrl is not null and that we aren't duplicating definitions for Url. I did not change the descriptions as I feel that would be make it slightly confusing since not all descriptions, labels, notes are the same.

---

I had to revert this change back to the original. Turns out this will break wingetcreate when it tries to generate the model based on the schema. The library we use to generate the model doesn't fully support schema keywords like "allOf" and "oneOf" so its better to leave it as is.